sql/opt/exec/execbuilder: TestExecBuild failed

sql/opt/exec/execbuilder.TestExecBuild [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3602508&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3602508&tab=artifacts#/) on master @ [649a8ca38cdc0883dd7a7811b189ea53af19f644](https://github.com/cockroachdb/cockroach/commits/649a8ca38cdc0883dd7a7811b189ea53af19f644):


```
            │ KV bytes read: 608 B
            │ estimated row count: 5
            │ count: 5
            │
            └── • lookup join
                │ nodes: <hidden>
                │ regions: <hidden>
                │ actual row count: 5
                │ KV time: 0µs
                │ KV contention time: 0µs
                │ KV rows read: 76
                │ KV bytes read: 608 B
                │ estimated row count: 100
                │ table: b@b_k_s_idx
                │ equality: (j) = (k)
                │
                └── • scan
                      nodes: <hidden>
                      regions: <hidden>
                      actual row count: 76
                      KV time: 0µs
                      KV contention time: 0µs
                      KV rows read: 76
                      KV bytes read: 608 B
                      estimated row count: 100 - 1,001 (100% of the table; stats collected <hidden> ago)
                      table: a@a_i_j_idx
                      spans: FULL SCAN
        goroutine 96 [running]:
        runtime/debug.Stack(0xc020ae5270, 0xa7a358200, 0x58ff020)
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x9f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin(0x59a68b8, 0xc00077c180, 0xc020ae5270, 0xa7a358200)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60 +0x7b
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x59a68b8, 0xc00077c180, 0xc020ae5270)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:41 +0x65
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processSubtest(0xc00030c400, 0x0, 0x0, 0xc01b870690, 0x0, 0xc00076a790, 0xe, 0x4461f2c, 0x5, 0x1, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:2311 +0x2230
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).processTestFile(0xc00030c400, 0xc00076a790, 0xe, 0x4461f2c, 0x5, 0x1, 0x0, 0x445f367, 0x3, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:1856 +0x7b8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.(*logicTest).runFile(0xc00030c400, 0xc00076a790, 0xe, 0x4461f2c, 0x5, 0x1, 0x0, 0x445f367, 0x3, 0x0, ...)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3204 +0xd8
        github.com/cockroachdb/cockroach/pkg/sql/logictest.RunLogicTestWithDefaultConfig.func1.1(0xc00077c180)
        	/go/src/github.com/cockroachdb/cockroach/pkg/sql/logictest/logic.go:3410 +0x2de
        testing.tRunner(0xc00077c180, 0xc00099d4c0)
        	/usr/local/go/src/testing/testing.go:1193 +0xef
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1238 +0x2b3
--- done: testdata/limit with config local: 27 tests, 0 failures
        --- FAIL: TestExecBuild/local/limit (48.29s)
=== RUN   TestExecBuild/local
    --- FAIL: TestExecBuild/local (0.00s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestExecBuild PKG=./pkg/sql/opt/exec/execbuilder TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
/cc @cockroachdb/sql-queries RaduBerinde
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/opt/exec/execbuilder/testdata/limit
+++ b/pkg/sql/opt/exec/execbuilder/testdata/limit
@@ -512,48 +512,21 @@ statement ok
 ANALYZE b;
 
 query T retry
-EXPLAIN ANALYZE SELECT * FROM a INNER LOOKUP JOIN b ON k = j ORDER BY i LIMIT 5;
+EXPLAIN SELECT * FROM a INNER LOOKUP JOIN b ON k = j ORDER BY i LIMIT 5;
 ----
-planning time: 10µs
-execution time: 100µs
-distribution: <hidden>
-vectorized: <hidden>
-rows read from KV: 176 (1.4 KiB)
-maximum memory usage: <hidden>
-network usage: <hidden>
-regions: <hidden>
+distribution: local
+vectorized: true
 ·
 • limit
-│ nodes: <hidden>
-│ regions: <hidden>
-│ actual row count: 5
-│ KV time: 0µs
-│ KV contention time: 0µs
-│ KV rows read: 76
-│ KV bytes read: 608 B
 │ estimated row count: 5
 │ count: 5
 │
 └── • lookup join
-    │ nodes: <hidden>
-    │ regions: <hidden>
-    │ actual row count: 5
-    │ KV time: 0µs
-    │ KV contention time: 0µs
-    │ KV rows read: 76
-    │ KV bytes read: 608 B
     │ estimated row count: 100
     │ table: b@b_k_s_idx
     │ equality: (j) = (k)
     │
     └── • scan
-          nodes: <hidden>
-          regions: <hidden>
-          actual row count: 100
-          KV time: 0µs
-          KV contention time: 0µs
-          KV rows read: 100
-          KV bytes read: 800 B
           estimated row count: 100 - 1,001 (100% of the table; stats collected <hidden> ago)
           table: a@a_i_j_idx
           spans: FULL SCAN

